src/main/java/io/aiven/kafka/connect/transforms/ExtractTimestampConfig.java

```diff
@@ -16,34 +16,109 @@
 
 package io.aiven.kafka.connect.transforms;
 
+import java.util.Arrays;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
 
 import org.apache.kafka.common.config.AbstractConfig;
 import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
 
 final class ExtractTimestampConfig extends AbstractConfig {
+
     public static final String FIELD_NAME_CONFIG = "field.name";
-    private static final String FIELD_NAME_DOC =
-        "The name of the field is to be used as the source of timestamp. "
+    private static final String FIELD_NAME_DOC = "The name of the field is to be used as the source of timestamp. "
             + "The field must have INT64 or org.apache.kafka.connect.data.Timestamp type "
             + "and must mot be null.";
 
+    public static final String EPOCH_RESOLUTION_CONFIG = "timestamp.resolution";
+    private static final String EPOCH_RESOLUTION_DOC = "Time resolution used for INT64 type field. "
+            + "Valid values are \"seconds\" for seconds since epoch and \"milliseconds\" for "
+            + "milliseconds since epoch. Default is \"milliseconds\" and ignored for "
+            + "org.apache.kafka.connect.data.Timestamp type.";
+
+
+    public enum TimestampResolution {
+
+        MILLISECONDS("milliseconds") {
+            @Override
+            long convert(final long timestamp) {
+                return timestamp;
+            }
+        },
+        SECONDS("seconds") {
+            @Override
+            long convert(final long timestamp) {
+                return TimeUnit.MICROSECONDS.toSeconds(timestamp);
```

Comment:
Must be `MILLISECONDS` (and in some other places)

Reply:
Also, vice versa: `TimeUnit.SECONDS.toMillis(timestamp)`. Better rename the parameter to something like `timestampInSeconds`, or just `seconds`. Because timestamp normally implies milliseconds.

Reply:
True, misunderstood idea behind.